### DISCUSSION TITLE
`FileSystemWatcher` drops events for nested files on Codespaces

### DISCUSSION THREAD

**Discussion by mattseddon** [microsoft/vscode-discussions#486] [Extension Development QnA] [answered]
### Tl;dr:

On Codespaces the extension host's `FileSystemWatcher` seems to drop events for files that are at a certain depth within a nested folder structure. Here is a quick demo of what I am talking about (with log level set to trace and the "Output" tab showing the "Remote Server"):

https://user-images.githubusercontent.com/37993418/219987038-f5ab8b90-5bac-410c-8957-fa7b70271257.mov

I cannot recreate behaviour outside of Codespaces. I.e locally or inside of a normal devcontainer. Is this normal/expected or is there an issue here?

### More details:

Hi, 

I work at [iterative.ai](https://iterative.ai/) and we've built a VS Code extension ([DVC for VS Code](https://marketplace.visualstudio.com/items?itemName=Iterative.dvc)). One of the main use cases is that you can run, compare, visualize, and track machine learning experiments right in VS Code. The extension can provide real-time updates to pre-defined plots. In order to do this we rely on the extension host's `FileSystemWatcher`. That looks like this:

https://user-images.githubusercontent.com/37993418/220808495-195d85ac-1ebb-45e6-89ca-1278a22d687d.mov

We got some reports that "plots live updates" stopped working in Codespaces from our customer success team (https://github.com/iterative/vscode-dvc/issues/3264). Investigating further it appears that the `FileSystemWatcher`'s behaviour is different in Codespaces.

Please let me know if there is a better place to raise this issue and I will happily move it there. 

Thank you.

**Comment by isidorn** [comment, marked as answer]
@mattseddon sorry for the slow response. This sounds like a bug to me. I suggest to open a new issue in the VS Code repo and we can include the Codespaces team if needed. Thank you
